### ISSUE TITLE
synchronization of transforms in the scene tree and b2world.

### ISSUE THREAD

**Issue by jordo** [briansemrau/godot_box2d#14]
I may play around a bit with this

```
void Box2DPhysicsBody::state_changed() {
	set_block_transform_notify(true);
	set_global_transform(b2_to_gd(body->GetTransform()));
	//if (get_script_instance())
	//	get_script_instance()->call("_integrate_forces");
	set_block_transform_notify(false);

	// TODO something? check rigidbody2d impl
	//if (contact_monitoring) {
	//	world_node->world.contac
	//}
}
```

Will definitely want to be able 'zoom' into, and rotate the view of the box2d world?   (Maybe putting the world under a node that is scale by 2, or even rotated or whatever.  If we set global transforms as the sync to tree from b2d, we're going to lose that.

How to handle in-tree transformations as well?  What happens if I added a Node2D under Box2DWorld, scale and rotate it, then add Box2dBodies under that node?   What if I animate that node running as well?  

And maybe the best first solution is to just punt on any in-tree transformations between the Box2dWorld and Box2dBody nodes, but stuff above the Box2dWorld node should definitely work.

**Comment by jordo** [comment]
So after discussion, it seems scale can't be supported anywhere in-between the node holding the b2world (Box2DWorld), and a node holding a b2body (Box2dPhysicsBody).  And likewise between a body and a fixture.

**Comment by jordo** [comment]
I'm anyways, I'm unsure if that's inclusive of Box2DWorld.    I still wanna try and a do a relative to world transform, as I think this would be better and even more flexible than whats currently in godot

**Comment by briansemrau** [comment]
@jordo Is this topic closed by #20 ?

**Comment by jordo** [comment]
You can close this as well.  but I will create and keep another issue open to refactor this once https://github.com/godotengine/godot/pull/44200 is merged upstream

The current implementation will break whenever a non-Node2D is in-between the Box2dWorld and a Box2dPhysicsBody.  With the upstream pull, I can make it cleaner, and support any CanvasItem between the World and Bodies.

**Comment by jordo** [comment]
As far as I know this is solved by #20, closing
